src/supervisor.tex

```diff
@@ -1726,16 +1739,31 @@ \subsection{Addressing and Memory Protection}
 The PTE format for Sv39 is shown in Figure~\ref{sv39pte}.  Bits 9--0
 have the same meaning as for Sv32.
 
-The C bit indicates that a PTE uses a custom implementation-specific encoding.
-If bit 63 is set, the algorithm for virtual-to-physical address translation
-is implementation-defined.  If bit 63 is not set, the algorithm for
-virtual-to-physical address translation is described in
-Section~\ref{napotpages}.
+\begin{table}[h]
+\begin{center}
+\begin{tabular}{|c|c||l|}
+\hline
+C & V & Description \\
+\hline
+0 & 0 & The PTE is invalid; all bits except C are don't cares and can be used freely by software \\
+0 & 1 & The PTE is valid and uses standard address-translation rules \\
+1 & 0 & {\em Reserved for standard use} \\
```

Comment:
This is a backward incompatible change.  The ratified spec has:

> The V bit indicates whether the PTE is valid; if it is 0, all other bits in the PTE are don’t-cares and may be used freely by software.

On Linux you would need to create a 64 EiB swap file (may be slightly off) to create a PTE with the high bit set, but there may be other supervisors that can create such PTEs under realistic circumstances.

Reply:
When V=1, bit 63 is currently reserved and must be zeroed by software for forward compatibility.  The PPN field doesn't automatically expand to fill all the upper bits.  So I don't think this one is backwards-incompatible.

When V=0, yes, we would be changing the meaning of bit 63 from don't care to reserved, which is technically a backwards-incompatible change.  We don't currently have a specific plan for the V=0, C=1 combination now marked reserved, so I don't see a problem if we wanted to change that C bit back to don't care when V=0 and maintain compatibility.

Reply:
I am referring only to the V=0 C=1 combination, which is non-reserved in the ratified spec.

Reply:
This table was a mistake on my part; apologies.  Hopefully the new table (with C and N) makes more sense.

Reply:
New table version looks good to me.